ACadSharp/IO/Templates/DwgInsertTemplate.cs

```diff
@@ -33,7 +33,32 @@ public override void Build(CadDocumentBuilder builder)
 
 			if (this.BlockHeaderHandle.HasValue)
 			{
-				insert.Block = builder.GetCadObject<Block>(this.BlockHeaderHandle.Value);
+				// orig code
+				// insert.Block = builder.GetCadObject<Block>(this.BlockHeaderHandle.Value);
+
+				// problem: builder.GetCadObject<Block>(this.BlockHeaderHandle.Value) returns null
+				// because, in the case of an Insert object, GetObject fails the type (Block) check.
+				// the templates' CadObject is not of type "Block"
+
+				// proposed solution:
+				// 1) extract the handle	
+				ulong handle = this.BlockHeaderHandle.Value;
+
+				// 2) attempt to use the original GetCadObject
+				Block bl = builder.GetCadObject<Block>(handle);
+
+				if (bl != null) 
+				{
+					// 3) if the original GetCadObject returns a Block, use it
+					insert.Block = bl;
+				} 
+				else 
+				{
+					// 4) if the original GetCadObject returns null, use the BlockEntity
+					// from the BlockRecordTemplate
+					DwgBlockRecordTemplate brt = builder.GetObjectTemplate<DwgBlockRecordTemplate>(handle);
+					insert.Block = brt.CadObject.BlockEntity;
```

Comment:
The ```Block``` field in the ```ACadSharp.Entitites.Insert``` should be replaced for a ```BlockRecord``` type, as you can see in your fix you are getting a BlockRecord template form the builder, so the insert should be referenced to the BlockRecord not the entity in it.